packages/mailgun/src/rest/v1/emails/emails-api.ts

```diff
@@ -0,0 +1,214 @@
+   */
+  public async sendMimeEmail(domainName: string, request: SendMimeEmailRequest): Promise<SendEmailResponse> {
+    this.client = this.getSinchClient();
+    const getParams = this.client.extractQueryParams<SendMimeEmailRequest>(request, [] as never[]);
+    const headers: { [key: string]: string | undefined } = {
+      Accept: 'application/json',
+    };
+    const body: RequestBody = transformSendMimeEmailRequestIntoApiRequestBody(request);
+    const basePathUrl = `${this.client.apiClientOptions.hostname}/v3/${domainName}/messages.mime`;
+
+    const requestOptions = await this.client.prepareOptions(basePathUrl, 'POST', getParams, headers, body || undefined);
+    const url = this.client.prepareUrl(requestOptions.hostname, requestOptions.queryParams);
+
+    const apiResponse = await this.client.processCall<SendEmailResponseFromApi>({
+      url,
+      requestOptions,
+      apiName: this.apiName,
+      operationId: 'sendMimeEmail',
+    });
+
+    return transformSendEmailResponseIntoClientResponse(apiResponse);
+  }
+
+  /**
+   * Retrieve a stored email
+   * Event(s) created from sending an email with Mailgun will contain a `storage.key` to use to retrieve the email.
+   * @param { string } domainName - Domain name that was used to send the email
+   * @param { string } storageKey - Storage key from the emails associated events
+   */
+  public async getEmail(domainName: string, storageKey: string): Promise<GetEmailResponse> {
+    this.client = this.getSinchClient();
+    const getParams = {};
+    const headers: { [key: string]: string | undefined } = {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    };
+    const body: RequestBody = '';
+    const basePathUrl = `${this.client.apiClientOptions.hostname}/v3/domains/${domainName}/messages/${storageKey}`;
+
+    const requestOptions = await this.client.prepareOptions(basePathUrl, 'GET', getParams, headers, body || undefined);
+    const url = this.client.prepareUrl(requestOptions.hostname, requestOptions.queryParams);
+
+    const apiResponse = await this.client.processCall<GetEmailResponseFromApi>({
+      url,
+      requestOptions,
+      apiName: this.apiName,
+      operationId: 'getEmail',
+    });
+
+    return transformGetEmailResponseIntoClientResponse(apiResponse);
+  }
+
+  /**
+   * Delete scheduled and undelivered mail
+   * Deletes all scheduled and undelivered mail from the domain queue.  This endpoint must be called on the storage API host and in the domain's region.  e.g. https://storage-us-east4.api.mailgun.net/v3/example.com/envelopes
+   *
+   * The storage hosts are `storage-us-east4.api.mailgun.net`, `storage-us-west1.api.mailgun.net`, and `storage-europe-west1.api.mailgun.net`.
+   * @param { string } domainName - The name of the domain you want to delete envelope from
+   */
+  public async purgeDomainQueues(domainName: string): Promise<GenericResponse> {
+    const requests = this.storageHostnames.map((hostname) =>
```

Comment:
We should find a better way to not send multiple request and not sending request outside of EU if located EU.
When https://documentation.mailgun.com/docs/mailgun/api-reference/openapi-final/tag/Events/#tag/Events/operation/get-v3-domain_name-events will be supported we could reduce to a single URL call

Reply:
Thinking about it, I'm not sure this method will be helpful as we must be sure is stores all the events related to the scheduled emails long enough and the resulting operation can take some time, depending on the number of pages to fetch, the availability of the API, the network speed...
With the approach I have implemented, we don't care about any of that.